### PR TITLE
[15.0][IMP] stock_barcodes: Skip backorder creation for pickings with all moves read or ignore pending quantities

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @tecnativa 

ping @carlosdauden @chienandalu

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

File: stock_barcodes/models/stock_picking.py [hunk 1]
```diff
@@ -46,7 +47,38 @@ def button_validate(self):
...
+                    < 0
+                ):
+                    # In normal conditions backorder will be created
+                    create_backorder = True
+                    if not move.move_line_ids or any(
+                        sml.state in ["pending"] for sml in move.move_line_ids
+                    ):
+                        # If any move are not processed we can not skip backorder
+                        skip_backorder = False
+                        break
```
Comment (pedrobaeza) [inline stock_barcodes/models/stock_picking.py:72]: Any line will impact in the whole picking?

**Review by pedrobaeza — COMMENTED** [review]
I miss regression tests for all these specific cases.

File: stock_barcodes/models/stock_picking.py (hunk 1 above)
Comment (pedrobaeza) [reply stock_barcodes/models/stock_picking.py:72]: @sergio-teruel can you answer this? The problem I see is that this aborts the evaluation of the rest of the picking. Is that correct?

Comment (pedrobaeza) [reply stock_barcodes/models/stock_picking.py:72]: ping @sergio-teruel

**Comment by sergio-teruel** [comment]
Yes, I know!!

**Comment by pedrobaeza** [comment]
OK, it seems weird...

/ocabot merge patch

Please include it in the v16 migration

**Pull request merged**